src/evaluation.py

```diff
@@ -0,0 +1,54 @@
+from lstm_probing_model import lstm_probing
+import torchtext
+import torch
+from torch import nn
+import numpy as np
+from torchtext.data.metrics import bleu_score
+import dataset
+from torch.utils.data import DataLoader
+
+def collate_fn_decode(data):
+    hidden_states, explanation = zip(*data)
+    return hidden_states, explanation
+
+def evaluate(tokenizer, model, encoder_states, exp_dataset, args):
+    training_dataset = dataset.TrainDataset(encoder_states, exp_dataset)
+    dataloader = DataLoader(training_dataset, batch_size=args.batch_size, collate_fn=collate_fn_decode)
+    model.eval()
+    overall_bleu = 0
+    decoded_words = []
+    reference_words = []
+    with torch.no_grad():
+        for batch, (x, y) in enumerate(dataloader):
+            x_tensor = torch.stack(list(x), dim=0).to(args.device)  # (batch, 768)
+            y_tensor = torch.stack(y, dim=0).to(args.device)  # (batch, 55)
+            for i in range(len(x_tensor)):
+                input_token = torch.tensor([[1]]).to(args.device)
```

Comment:
This looks weird because `input_token` is already a tensor, then you wrap it in a tensor again in L36. I think your code handles things fine when `input_token` is just a plain number (as is the case in future for loop iterations), so maybe just initialize `input_token` to just be the number 1 here?

Reply:
updated